config/site/site.config.js

```diff
@@ -0,0 +1,122 @@
+const i18nSupportedLocales = require('terra-i18n/lib/i18nSupportedLocales');
+const navConfig = require('./navigation.config');
+const startCase = require('lodash.startcase');
+const fs = require('fs');
+const path = require('path');
+
+const npmPackage = JSON.parse(fs.readFileSync(path.resolve(process.cwd(), 'package.json')));
+
+const siteConfig = {
+  /* The navigation configuration.  */
+  navConfig,
+
+  /* The path to the pages configuration. If this is enabled, page config will not be generated. */
+  // pagesConfig,
+
+  // These options are used to find the pages to serve via terra dev site.
+  // If 'pagesConfig' is provided, no pages will be generated.
+  // Output:
+  // {root}/lib/{entryPoint}/**/*{pageType}.{jsx,js,md,}
+  // {root}/packages/*/lib/{entryPoint}/**/*{pageType}.{jsx,js,md,}
+  // source will be used if hot reloading is enabled and not running in prod mode.
+  // customPatterns
+  // *NOTE* The entryPoints also act as the beginning of the directory strucure for menu navigation.
+  // For example: /root/src/<entryPointDir>/folder/item would start navigation at 'folder'.
+  generatePages: {
+    searchPatterns: [
+      {
+        root: process.cwd(),
+        source: 'src',
+        dist: 'lib',
+        entryPoint: 'terra-dev-site',
+      },
+    ],
+    customPatterns: [
+      // {
+      //   pattern: 'dir/lib/terra-dev-site/**/*.<type>.{jsx,js})',
+      //   entryPoint: path.join(process.cwd(),'lib', 'terra-dev-site'),
```

Comment:
Should the entry point in customPatterns behave the same as the search patterns? 
I think I would expect to be able to pass `root`, `source`, `dist` as well as `entryPoint` and `pattern`.

Reply:
I'm removing custom. With the non custom patterns, we're combining root, source/lib/entrypoint to make the pattern.